### ISSUE TITLE
[BUG] semantic error in learning objective loss function

### ISSUE THREAD

**Issue by le1nax** [MouseLand/cellpose#1299]
I think there is a sematic error in the learning objective loss function. in the train_seg function, at the part where you would compute mse between predicted flows and Gt flows:

`def _loss_fn_seg(lbl, y, device):
    """
    Calculates the loss function between true labels lbl and prediction y.

    Args:
        lbl (numpy.ndarray): True labels (cellprob, flowsY, flowsX).
        y (torch.Tensor): Predicted values (flowsY, flowsX, cellprob).
        device (torch.device): Device on which the tensors are located.

    Returns:
        torch.Tensor: Loss value.

    """
    criterion = nn.MSELoss(reduction="mean")
    criterion2 = nn.BCEWithLogitsLoss(reduction="mean")
    veci = 5. * lbl[:, -2:]
    loss = criterion(y[:, -3:-1], veci)
    loss /= 2.
    loss2 = criterion2(y[:, -1], (lbl[:, -3] > 0.5).float())
    loss = loss + loss2
    return loss
`

notice that in `veci = 5. * lbl[:, -2:]
    loss = criterion(y[:, -3:-1], veci)`, you compute mse bewteen GT and prediction. GT flows are multiplied by a factor of 5, while the prediction is not. Mathematically, you are multiplying only one element of the substraction and not the square of the substraction which I think would be correct. The paper claims to scale the mse to a factor of 5 to match the bce loss but I dont think this is the correct way.

**Comment by le1nax** [comment]
Btw Mediar did the same. So either they just blindly copied their semantic or I am missing something.

**Comment by mrariden** [comment]
@le1nax  I'm not sure where the paper says that the mse is scaled, but the relevant equation from the cellpose 1 paper is 

<img width="469" height="65" alt="Image" src="https://github.com/user-attachments/assets/0ffc2ae3-977c-4dd9-9648-0c4d554267c5" />

which is what is implemented in the code.

**Comment by marius10p** [comment]
"flow" does not have an intrinsic unit of measurement. Multiplying by 5 was just a choice of scaling, which we have kept in order to be backwards compatible. If you prefer, in your own code you can predict unscaled flows, or you can scale them (and the loss functions) however you like. The Cellpose flows will always be scaled by 5 for backwards compatibility.
